presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -159,7 +160,7 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
         Optional<HiveBucketHandle> hiveBucketHandle = shouldIgnoreTableBucketing ? Optional.empty() : getHiveBucketHandle(table);
         Optional<HiveBucketFilter> bucketFilter = shouldIgnoreTableBucketing ? Optional.empty() : getHiveBucketFilter(table, effectivePredicate);
 
-        if (hiveBucketHandle.isPresent() && hiveBucketHandle.get().getReadBucketCount() * partitions.size() > getMaxBucketsForGroupedExecution(session)) {
+        if (!queryUsesHiveBucketColumn(effectivePredicate) && hiveBucketHandle.isPresent() && getBucketCount(hiveBucketHandle.get(), bucketFilter) * partitions.size() > getMaxBucketsForGroupedExecution(session)) {
```

Comment:
nit: now that we have three conjunctive predicates, should we consider move `getBucketCount(hiveBucketHandle.get(), bucketFilter) * partitions.size() > getMaxBucketsForGroupedExecution(session)` into a separate method for readability, and put each predicate on a separate line?